### ISSUE TITLE
Fix request: false positive AvoidRecompilingXPathExpression

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#103]
example:
```
void bad() {
         XPathExpression expr = xpath.compile(xpathExpression);
 }
```
should have a violation for AvoidRecompilingXPathExpression and not for AvoidRecompilingPatterns. Also, 
AvoidRecompilingXPathExpression should only be triggered on XPath compile and not on another .compile.
It should also be clear/documented how to solve it.

**Comment by jborgers** [comment]
fix committed
